setup.cfg

```diff
@@ -45,3 +40,141 @@ ignore = E203,E501
 # D213 Multi-line docstring summary should start at the second line (reason: pep257 default)
 ignore = D107, D203, D213
 ignore_decorators = overrides
+
+[tox:tox]
+min_version = 4.0
+env_list =
+    # Parametrized environments.
+    # First parameter allows us to choose Python 3.8 or 3.10.
+    # Second parameter chooses how to define the environment:
+    #  withreq: using requirements-devel.txt
+    #  noreq: Without either requirements file (but including dev requirements)
+    # Third parameter selects the current unit tests (unit) or the legacy unit tests (legacy)
+    py{38,310}-{withreq,noreq}-{unit,legacy}
+skip_missing_interpreters = true
+labels =
+    # Minimal testing environments. run with `tox run-parallel -m test`
+    test = py38-withreq-{unit,legacy}
+    # Test in Python 3.10 from an empty environment.
+    future-test = py310-noreq-{unit,legacy}
+    # Environments to run in CI
+    ci = py{38,310}-withreq-{unit,legacy},py{38,310}-noreq-unit
+    # Just run the regular unit tests, not the legacy ones
+    unit = py{38,310}-{withreq,noreq}-unit
+    # Legacy unit tests only
+    legacy = py{38,310}-{withreq,noreq}-legacy
+set_env =
+    SNAPCRAFT_IGNORE_YAML_BINDINGS: "1"
+
+[testenv]
+deps =
+    withreq,pylint,mypy,pyright: -r{tox_root}/requirements-devel.txt
+    noreq: PyNaCl>=1.5.0
```

Comment:
If I understand correctly, `tox` always needs PyNaCl?  Or snapcraft needs it?

Reply:
Snapcraft was having breakages without PyNaCl. I'm not sure why as we don't explicitly require it. I would like at some point to tackle removing this and updating to the latest version of cryptography, but it was out of scope here. #3519 is the relevant PR where this was added, and I'm guessing that upgrading to the latest cryptography will let us remove this.

Reply:
I've created a task for this at CRAFT-1569